docs/json-argument-source.adoc

```diff
@@ -0,0 +1,192 @@
+:page-title: JSON Argument Source
```

Comment:
I renamed the file (and thus the URL) and changed the title to limit it to JSON argument sources - who knows what other fancy things we can do with JSON in the future that would otherwise name-clash. 😉